File: models/Gpt-3.5-turbo/py/code/introductory_problems/Apple_Division.py

n = int(input())
p = list(map(int, input().split()))

total_weight = sum(p)
half_weight = total_weight // 2

# Initialize the table
dp = [[float('inf')] * (half_weight + 1) for _ in range(n + 1)]
dp[0][0] = 0

for i in range(1, n + 1):
    for j in range(half_weight + 1):
        # Include the current apple
        if j >= p[i - 1]:
            dp[i][j] = min(dp[i][j], dp[i - 1][j - p[i - 1]])
        # Exclude the current apple
        dp[i][j] = min(dp[i][j], dp[i - 1][j])

# Minimum difference between the weights of the groups
min_diff = total_weight - 2 * dp[n][half_weight]
print(min_diff)

File: models/Gpt-3.5-turbo/cpp/code/problems/introductory_problems/Apple_Division.cpp

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
  int n;
  cin >> n;
  vector<int> p(n);
  int sum = 0;
  for (int i = 0; i < n; i++) {
    cin >> p[i];
    sum += p[i];
  }

  vector<vector<bool>> dp(n + 1, vector<bool>(sum / 2 + 1, false));
  dp[0][0] = true;

  for (int i = 1; i <= n; i++) {
    for (int j = 0; j <= sum / 2; j++) {
      dp[i][j] = dp[i - 1][j];
      if (j >= p[i - 1]) {
        dp[i][j] = dp[i][j] || dp[i - 1][j - p[i - 1]];
      }
    }
  }

  int minDiff = sum;
  for (int j = 0; j <= sum / 2; j++) {
    if (dp[n][j]) {
      minDiff = min(minDiff, sum - 2 * j);
    }
  }

  cout << minDiff << endl;

  return 0;
}
